assets/js/modules/pagespeed-insights/datastore/report.test.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * modules/pagespeed-insights data store: report tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { STORE_NAME } from './index';
+import {
+	createTestRegistry,
+	muteConsole,
+	subscribeUntil,
+	unsubscribeFromAll,
+} from 'tests/js/utils';
+import * as fixtures from './__fixtures__';
+
+describe( 'modules/pagespeed-insights report', () => {
+	const baseInfoVar = '_googlesitekitBaseData';
+	const baseInfo = {
+		referenceSiteURL: 'http://sitekit.10uplabs.com/',
+	};
+	const entityInfoVar = '_googlesitekitEntityData';
+	let apiFetchSpy;
+	let registry;
+	let store;
+
+	beforeAll( () => {
+		API.setUsingCache( false );
+	} );
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		store = registry.stores[ STORE_NAME ].store;
+
+		apiFetchSpy = jest.spyOn( { apiFetch }, 'apiFetch' );
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	afterEach( () => {
+		delete global[ baseInfoVar ];
+		delete global[ entityInfoVar ];
+		unsubscribeFromAll( registry );
+		apiFetchSpy.mockRestore();
+	} );
+
+	describe( 'actions', () => {
+
+	} );
+
+	describe( 'selectors', () => {
+		describe( 'getReport', () => {
+			it( 'uses a resolver to make a network request', async () => {
+				global[ baseInfoVar ] = baseInfo;
+				global[ entityInfoVar ] = {};
+
+				fetch
+					.doMockOnceIf(
+						/^\/google-site-kit\/v1\/modules\/pagespeed-insights\/data\/pagespeed/
+					)
+					.mockResponseOnce(
+						JSON.stringify( fixtures.desktopReport ),
+						{ status: 200 }
+					);
+
+				const options = { strategy: 'desktop', url: 'http://testsite.10uplabs.com/' };
+
+				const initialReport = registry.select( STORE_NAME ).getReport( options );
+
+				// Ensure the proper parameters were passed.
+				expect( fetch.mock.calls[ 0 ][ 0 ] ).toMatchQueryParameters(
+					options
+				);
+
+				expect( initialReport ).toEqual( undefined );
+				await subscribeUntil( registry,
+					() => (
+						registry.select( STORE_NAME ).getReport( options ) !== undefined
+					),
```

Comment:
I realize many of the older stores don't do this, but it's better to use the `hasFinishedResolution( selector, selectorArgs )` selector that's part of every store as this will fail faster if the selector is implemented improperly. It's also a bit more clear as to what you're waiting for 😄 
```suggestion
					() => registry.select( STORE_NAME ).hasFinishedResolution( 'getReport', [ options ] )
```